src/carousel.js

```diff
@@ -0,0 +1,284 @@
+    }
+};
+
+const deleteOldActiveSlide = () => {
+    const previousActiveElement = carouselContainer.querySelector(
+        '.carousel__slide--active'
+    );
+    if (previousActiveElement) {
+        previousActiveElement.classList.remove('carousel__slide--active');
+    }
+};
+
+const updateCurrentActiveSlide = slides => {
+    const middleElement = (maxCountOnScreen + 1) / 2;
+    const activeElement = slides[middleElement];
+    activeElement.classList.add('carousel__slide--active');
+};
+
+const setActiveSlide = slides => {
+    deleteOldActiveSlide();
+    updateCurrentActiveSlide(slides);
+};
+
+const updateActiveDotCounter = direction => {
+    if (direction === 'left') {
+        activeDot === 0 ? (activeDot = slides.length - 1) : activeDot--;
+    } else if (direction === 'right') {
+        activeDot === slidesCounter - 1 ? (activeDot = 0) : activeDot++;
+    }
+};
+
+const deleteOldActiveDot = () => {
+    const previousActiveDot = document.querySelector('.dotsNav__dot--active');
+    if (previousActiveDot) {
+        previousActiveDot.classList.remove('dotsNav__dot--active');
+    }
+};
+
+const addCurrentActiveDot = () => {
+    dots[activeDot].classList.add('dotsNav__dot--active');
+};
+
+const updateDots = () => {
+    deleteOldActiveDot();
+    addCurrentActiveDot();
+};
+
+const setActiveDot = direction => {
+    updateActiveDotCounter(direction);
+    updateDots();
+};
+
+const updateSlideOrder = () => {
+    slides = carouselContainer.querySelectorAll('.carousel__slide');
+};
+
+const setAcitveElements = (slides, direction) => {
+    setActiveSlide(slides);
+    setActiveDot(direction);
+};
+
+const moveToDirection = direction => {
+    if (!isInTransition) {
+        changeCarouselChildrenOrder(direction);
+        updateSlideOrder();
+        moveCarousel(slides, elementWidth);
+        setAcitveElements(slides, direction);
+    }
+};
+
+const moveAfterTransitionRight = () => {
+    moveToDirection('right');
+    carouselContainer.removeEventListener(
+        'transitionend',
+        moveAfterTransitionRight
+    );
+};
+
+const moveAfterTransitionLeft = () => {
+    moveToDirection('left');
+    carouselContainer.removeEventListener(
+        'transitionend',
+        moveAfterTransitionLeft
+    );
+};
+
+const checkMultipliersDiffrence = slide => {
+    // It is an integer that specifies the multiplier for the style 'left' to the beginning of the middle element.
+    // For example, the first visible element has this multiplier set to 0, because its 'left' style is the result of (multiplier * elementWidth), which is 0.
+    const activeElementMultiplier = Math.trunc(maxCountOnScreen / 2);
+    // Subtraction of active multiplier from current clicked slide multiplier.
+    const subtractionOfMultipliers =
+        parseFloat(slide.style.left) / elementWidth - activeElementMultiplier;
+    return subtractionOfMultipliers;
+};
+
+const isApplicableForAddingClass = slide => {
```

Comment:
jeśli funkcja ma zwracać booleana - to powinna zwracać `true` lub `false`
tutaj zwraca `true` albo nic :)
```js
const isApplicableForAddingClass = slides => {
  return checkMultipliersDiffrence(slide) === 0;
}
```
zwróci booleana